### DISCUSSION TITLE
Having trouble settting up TOTP secret key for login

### DISCUSSION THREAD

**Discussion by hartzburg** [mrlt8/docker-wyze-bridge#466] [Q&A]
Hello! It may be because I am new to Docker, but I am having trouble getting the bridge to auto login using the TOTP secret key. I am running this in a docker container on Ubuntu Server. I've got everything setup so that it runs, and I get the cams to stream if I manually insert the code using "docker exec -it wyze-bridge sh -c 'echo "123456" > /tokens/mfa_token.txt' ". I have obtained a new secret key and am trying to set it up such that the TOTP code logs in on its own. I am putting the key in a file in the linux directory "/tokens/totp" , but it seems to still be waiting for the /tokens/mfa_token.txt to be populated. Is the "/tokens/totp" directory located inside the docker container? If so, which is the best way to put the secret key in there such that it generates on its own? Again, this is more my in-experience with docker, so any help is appreciated. Thanks!

**Comment by mrlt8** [comment]
Yes, the bridge looks for the totp file inside the container (`/tokens/totp`), so you'd need to mount it within the container with something like this in your docker-compose.yml:

```yaml
volumes:
    - /path/to/my/totp_file:/tokens/totp
```

**Reply by hartzburg** [reply]
Excellent, thanks so much, that worked! I am now getting the continuous prompt "wyze-bridge    | 2022/07/07 01:44:16 [WyzeBridge] Non-base32 digit found"
"...
wyze-bridge    | 2022/07/07 01:44:15 [WyzeBridge] ?? Fetching 'auth' from the Wyze API...
wyze-bridge    | 2022/07/07 01:44:16 [WyzeBridge] ? MFA Token Required
wyze-bridge    | 2022/07/07 01:44:16 [WyzeBridge] Non-base32 digit found"

Is there a special format the secret key needs to be in? Or is it exactly as shown on the wyze app screen, i.e including dashes ?

Thanks for your help.

**Reply by mrlt8** [reply]
Hmm, there shouldn't be any dashes. The key that you need to copy and paste into the `/tokens/totp` file should be 16 characters and look like this:
![IMG_4821](https://user-images.githubusercontent.com/67088095/177910542-9f1f8fa2-5351-4da3-b4a6-f1f834f7fb68.jpg)

**Reply by KhaosMonkey** [reply]
> Yes, the bridge looks for the totp file inside the container (`/tokens/totp`), so you'd need to mount it within the container with something like this in your docker-compose.yml:
> 
> ```yaml
> volumes:
>     - /path/to/my/totp_file:/tokens/totp
> ```

How would one go about doing this with the unraid docker container?

**Reply by KhaosMonkey** [reply]
> > Yes, the bridge looks for the totp file inside the container (`/tokens/totp`), so you'd need to mount it within the container with something like this in your docker-compose.yml:
> > ```yaml
> > volumes:
> >     - /path/to/my/totp_file:/tokens/totp
> > ```
> 
> How would one go about doing this with the unraid docker container?

I Figured out that it requires you to not have a file extension like ".txt" on the file. so the file is legitimately named "totp"
